### PR TITLE
Building an object detection application with tensorflow.js and react.js on Ubuntu 20.04

### LINKED ISSUE #1995: Building an Object Detection Application with TensorFlow.js and React.js
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

Building an Object Detection Application with TensorFlow.js and React.js

### Introduction paragraph (2-3 paragraphs):

In this article, we will learn on how to build a real-time object detection application using TensorFlow.js and React.js.

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this article, the reader should be able to:

1. Access the React.JS and Tensorflow.JS Computer Vision Template
2. Install Tensorflow.JS and setup the pre-built models
3. Build a React.JS app that accesses the webcam
4. Make detection from the webcam in real time

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by wmkinyan**
This fixes #1995

### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import { drawRect } from "./utilities";
+```
+
+The next thing that we are going to do is to load our network into our model. This is achieved by issuing the following command:
+
+```bash
+    const net = await cocossd.load();
+```
+
+We first create a new variable called 'net'. Because our function is asynchronous, we wait for the 'cocossd' model to load.
```
Comment (collins-a) [inline articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md:87]: What would happen if our function would be of a synchronous paradigm? What is the significance of the asynchronous function in this context?

File: articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+```
+
+It is then executed with the command:
+
+```bash
+    drawRect(obj, ctx);
+```
+This enables bounding boxes to be drawn on the video.
+
+To access the complete code, try and test the code out for yourself, you can find the complete code [here](https://replit.com/@wmkinyan/RealTimeObjectDetectionTFJSReact#README.md).
```
Comment (collins-a) [inline articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md:138]:
```suggestion
To access and tinker with the complete block of code, you can find it [here](https://replit.com/@wmkinyan/RealTimeObjectDetectionTFJSReact#README.md).
```

File: articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+### Conclusion
+
+That wraps it all! 
+
+In this tutorial, we have learned how to build an object detection application using the TensorFlow library in the React app.
```
Comment (collins-a) [inline articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md:144]:
```suggestion
That wraps it up! In this tutorial, we have learned how to build an object detection application using the TensorFlow library in the React app.
```

**Review by collins-a — COMMENTED** [review]
@wmkinyan very useful article. A few changes before we can continue with the tech review

**Review by collins-a — APPROVED** [review]
Hey @wmkinyan. It runs on my end. I'm satisfied with the changes. However, hopefully next time you can have us test out the program on replit.

**Comment by collins-a** [comment]
Hey @wmkinyan.

Just bringing to your attention that the moment our EngEd repo has a new structure. As such, articles are getting migrated to a new Articles folder. To make this process as easy as possible, and ensure that your article is published as soon as possible:

- Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
- Delete your article directory from the articles folders.
- Do a git pull (take a look at the new structure of the repo after a git pull).
- Upload your article files to the [new articles folder, inside content](https://github.com/section-engineering-education/engineering-education/tree/master/content/articles).

Ping me via Slack if you have any problems.

**Review by collins-a — APPROVED** [review]

**Comment by smkinyan** [comment]
Hi @hectorkambow As I explained in the slack channel, the duplicated version #2350 is the one to be published and not this PR #2135

**Comment by hectorkambow** [comment]
> Hi @hectorkambow As I explained on the slack channel, the duplicated version #2350 is the one to be published and not this PR #2135

Thank you 👍

**Pull request closed without merging**